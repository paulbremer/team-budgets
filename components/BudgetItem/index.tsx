import { Text, StyleSheet, View } from "react-native";
import React from "react";
import ProgressBar from "react-native-progress/Bar";
import { useThemeColor } from "@/hooks/useThemeColor";
import { Stack, HStack } from "react-native-flex-layout";

import { BudgetType } from "@/types/budgets";

type BudgetItemProps = {
  budget: BudgetType;
};

export default function BudgetItem({ budget }: BudgetItemProps) {
  if (!budget) return;

  const tintColor = useThemeColor({}, "tint");
  const { title, remainingBudget, total, startDate, endDate } = budget;

  return (
    <View style={styles.container}>
      <Stack spacing={8}>
        <Text style={styles.title}>ðŸ’° {title}</Text>
        <HStack justify="between">
          <Text>Remaining budget</Text>
          <Text>
            &euro; {remainingBudget} / &euro; {total}
          </Text>
        </HStack>
        <ProgressBar progress={remainingBudget / total} width={null} height={10} color={tintColor} borderRadius={4} />
        <HStack justify="between" mt={8}>
          <Text>Period</Text>
          <Text>
            {startDate.toLocaleDateString()} - {endDate.toLocaleDateString()}
          </Text>
        </HStack>
      </Stack>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    marginVertical: 8,
    padding: 20,
    borderRadius: 8,
  },
  title: {
    fontWeight: "bold"
  },
});
