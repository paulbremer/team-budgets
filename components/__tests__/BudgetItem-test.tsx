import * as React from "react";
import renderer from "react-test-renderer";

import BudgetItem from "../BudgetItem";

const mockBudget = {
  id: '1',
  title: "test budget",
  startDate: new Date(),
  endDate: new Date(),
  total: 2000,
  remainingBudget: 1000,
};

it(`renders correctly`, () => {
  const tree = renderer.create(<BudgetItem budget={mockBudget} />).toJSON();

  expect(tree).toMatchSnapshot();
});
