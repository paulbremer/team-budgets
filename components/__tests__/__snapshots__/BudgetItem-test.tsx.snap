// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "marginVertical": 8,
      "padding": 20,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": undefined,
          "borderBottomColor": undefined,
          "borderBottomEndRadius": undefined,
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderBottomStartRadius": undefined,
          "borderBottomWidth": undefined,
          "borderColor": undefined,
          "borderEndColor": undefined,
          "borderEndWidth": undefined,
          "borderLeftColor": undefined,
          "borderLeftWidth": undefined,
          "borderRadius": undefined,
          "borderRightColor": undefined,
          "borderRightWidth": undefined,
          "borderStartColor": undefined,
          "borderStartWidth": undefined,
          "borderStyle": undefined,
          "borderTopColor": undefined,
          "borderTopEndRadius": undefined,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "borderTopStartRadius": undefined,
          "borderTopWidth": undefined,
          "borderWidth": undefined,
          "bottom": undefined,
          "end": undefined,
          "height": undefined,
          "left": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "opacity": undefined,
          "overflow": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingEnd": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingStart": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "start": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        },
        [
          {
            "alignContent": undefined,
            "alignItems": undefined,
            "alignSelf": undefined,
            "flex": undefined,
            "flexBasis": undefined,
            "flexDirection": undefined,
            "flexGrow": undefined,
            "flexShrink": undefined,
            "flexWrap": undefined,
            "justifyContent": undefined,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontWeight": "bold",
          },
          [
            {
              "marginBottom": 8,
            },
            undefined,
          ],
        ]
      }
    >
      ðŸ’° 
      test budget
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": undefined,
            "borderBottomColor": undefined,
            "borderBottomEndRadius": undefined,
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderBottomStartRadius": undefined,
            "borderBottomWidth": undefined,
            "borderColor": undefined,
            "borderEndColor": undefined,
            "borderEndWidth": undefined,
            "borderLeftColor": undefined,
            "borderLeftWidth": undefined,
            "borderRadius": undefined,
            "borderRightColor": undefined,
            "borderRightWidth": undefined,
            "borderStartColor": undefined,
            "borderStartWidth": undefined,
            "borderStyle": undefined,
            "borderTopColor": undefined,
            "borderTopEndRadius": undefined,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "borderTopStartRadius": undefined,
            "borderTopWidth": undefined,
            "borderWidth": undefined,
            "bottom": undefined,
            "end": undefined,
            "height": undefined,
            "left": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginEnd": undefined,
            "marginHorizontal": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginStart": undefined,
            "marginTop": undefined,
            "marginVertical": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "opacity": undefined,
            "overflow": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingEnd": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingStart": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "start": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          },
          [
            {
              "alignContent": undefined,
              "alignItems": undefined,
              "alignSelf": undefined,
              "flex": undefined,
              "flexBasis": undefined,
              "flexDirection": "row",
              "flexGrow": undefined,
              "flexShrink": undefined,
              "flexWrap": undefined,
              "justifyContent": "space-between",
            },
            [
              undefined,
              [
                {
                  "marginBottom": 8,
                },
                undefined,
              ],
            ],
          ],
        ]
      }
    >
      <Text
        style={
          [
            undefined,
            [
              {
                "marginEnd": 0,
              },
              undefined,
            ],
          ]
        }
      >
        Remaining budget
      </Text>
      <Text
        style={
          [
            undefined,
            [
              null,
              undefined,
            ],
          ]
        }
      >
        â‚¬ 
        1000
         / â‚¬ 
        2000
      </Text>
    </View>
    <View
      animated={true}
      animationConfig={
        {
          "bounciness": 0,
        }
      }
      animationType="spring"
      indeterminate={false}
      indeterminateAnimationDuration={1000}
      onLayout={[Function]}
      progress={0.5}
      style={
        [
          {
            "backgroundColor": undefined,
            "borderColor": "#0a7ea4",
            "borderRadius": 4,
            "borderWidth": 1,
            "overflow": "hidden",
            "width": null,
          },
          [
            undefined,
            [
              {
                "marginBottom": 8,
              },
              undefined,
            ],
          ],
        ]
      }
      useNativeDriver={false}
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#0a7ea4",
            "height": 10,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateX": 0.5,
              },
              {
                "scaleX": 0.50005,
              },
            ],
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "backgroundColor": undefined,
            "borderBottomColor": undefined,
            "borderBottomEndRadius": undefined,
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderBottomStartRadius": undefined,
            "borderBottomWidth": undefined,
            "borderColor": undefined,
            "borderEndColor": undefined,
            "borderEndWidth": undefined,
            "borderLeftColor": undefined,
            "borderLeftWidth": undefined,
            "borderRadius": undefined,
            "borderRightColor": undefined,
            "borderRightWidth": undefined,
            "borderStartColor": undefined,
            "borderStartWidth": undefined,
            "borderStyle": undefined,
            "borderTopColor": undefined,
            "borderTopEndRadius": undefined,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "borderTopStartRadius": undefined,
            "borderTopWidth": undefined,
            "borderWidth": undefined,
            "bottom": undefined,
            "end": undefined,
            "height": undefined,
            "left": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginEnd": undefined,
            "marginHorizontal": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginStart": undefined,
            "marginTop": 8,
            "marginVertical": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "opacity": undefined,
            "overflow": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingEnd": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingStart": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "start": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          },
          [
            {
              "alignContent": undefined,
              "alignItems": undefined,
              "alignSelf": undefined,
              "flex": undefined,
              "flexBasis": undefined,
              "flexDirection": "row",
              "flexGrow": undefined,
              "flexShrink": undefined,
              "flexWrap": undefined,
              "justifyContent": "space-between",
            },
            [
              undefined,
              [
                null,
                undefined,
              ],
            ],
          ],
        ]
      }
    >
      <Text
        style={
          [
            undefined,
            [
              {
                "marginEnd": 0,
              },
              undefined,
            ],
          ]
        }
      >
        Period
      </Text>
      <Text
        style={
          [
            undefined,
            [
              null,
              undefined,
            ],
          ]
        }
      >
        10/4/2024
         - 
        10/4/2024
      </Text>
    </View>
  </View>
</View>
`;
